csrc/root_domain_map.cpp

```diff
@@ -197,6 +197,52 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
     return dom_map;
   }
 
+  if (LinearOp* op = dynamic_cast<LinearOp*>(consumer_tv_->definition())) {
+    auto out_size = consumer_root.size();
+
+    // Check if the producer is A, B or bias.
+    MatmulRole input_role;
+    if (producer->sameAs(op->inA()->as<TensorView>()->domain())) {
+      input_role = MatmulRole::INPUT_A;
+    } else if (producer->sameAs(op->inB()->as<TensorView>()->domain())) {
+      input_role = MatmulRole::INPUT_B;
+    } else {
+      input_role = MatmulRole::INPUT_C;
+    }
+
+    // LinearOp:
+    // inputs (INPUT_A) = {*, in_features}
+    // weight (INPUT_B) = {out_features, in_features} / {in_features}
+    // bias (INPUT_C) = {out_features} / {}
+    // output = {*, out_features} / {*}
+
+    switch (input_role) {
+      case MatmulRole::INPUT_A: {
+        // Linear output is same as input for all but the last dimension
+        for (auto inx : c10::irange(producer_root.size() - 1)) {
+          updatePairwiseRootDomainMap(
+              producer_root.at(inx), consumer_root.at(inx));
+        }
+        break;
+      }
+      case MatmulRole::INPUT_B: {
+        if (producer_root.size() == 1) {
+          // out_features is not present, no mapping required.
+          break;
+        }
+      }
+      case MatmulRole::INPUT_C: {
+        // The last dimension of LinearOp is out_features.
+        updatePairwiseRootDomainMap(
+            producer_root.at(0), consumer_root.at(out_size - 1));
+        break;
+      }
+      default:
+        NVF_ERROR("Unexpected input type.");
+    }
```

Comment:
Is there any value in factoring this out the way you did with `ops::mapMatmulOpIterDomains`?

Reply:
Yes, I'll factor it out now. This was a simple enough mapping so we didn't need the utility necessarily but given that we may make changes to the mapping for K dimension, it is useful to consolidate the changes.